### PR TITLE
Implemented the following Jax Frontend functions:

### LINKED ISSUE #5680: asarray #4612

### LINKED ISSUE #5681: cumsum #4612

### LINKED ISSUE #5682: cumprod #4612

### PR DESCRIPTION
**Pull request by AlphaPro06**
Implemented jax numpy frontend functions

### CODE REVIEW

File: .idea/ivy.iml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <module type="PYTHON_MODULE" version="4">
   <component name="NewModuleRootManager">
-    <content url="file://$MODULE_DIR$" />
-    <orderEntry type="jdk" jdkName="Remote Python 3.8.10 Docker (unifyai/ivy:latest)" jdkType="Python SDK" />
+    <content url="file://$MODULE_DIR$">
```
Comment (YushaArif99) [inline .idea/ivy.iml:4]: The contents of .idea folder should not be modified. Kindly remove these changes from the commit :)

File: .idea/misc.xml [hunk 2]
```diff
@@ -3,8 +3,8 @@
   <component name="JavaScriptSettings">
     <option name="languageLevel" value="ES6" />
   </component>
-  <component name="ProjectRootManager" version="2" project-jdk-name="Remote Python 3.8.10 Docker (unifyai/ivy:latest)" project-jdk-type="Python SDK" />
+  <component name="ProjectRootManager" version="2" project-jdk-name="Python 3.8 (ivy_dev) (2)" project-jdk-type="Python SDK" />
```
Comment (YushaArif99) [inline .idea/misc.xml:6]: Same is the case here as well

File: .idea/runConfigurations/_template__of_py_test.xml [hunk 3]
```diff
@@ -3,9 +3,9 @@
     <module name="ivy" />
     <option name="INTERPRETER_OPTIONS" value="" />
     <option name="PARENT_ENVS" value="true" />
-    <option name="SDK_HOME" value="docker://ivydl/ivy:latest/python3" />
-    <option name="WORKING_DIRECTORY" value="/ivy" />
-    <option name="IS_MODULE_SDK" value="false" />
+    <option name="SDK_HOME" value="" />
```
Comment (YushaArif99) [inline .idea/runConfigurations/_template__of_py_test.xml:6]: Same here as well

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_namespace_functions.py [hunk 4]
```diff
@@ -794,3 +794,566 @@ def test_jax_numpy_cos(
...
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        frontend="jax",
+        fn_tree="numpy.bitwise_or",
+        x1=x[0],
+        x2=x[1],
+    )
+
+
+# Testing ivy.numpy.bitwise_or functionality with signed integer arrays
```
Comment (YushaArif99) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_namespace_functions.py:832]: I don't see the point of writing separate test functions for different data types. Passing` valid` or `float` for the `get_dtype` helper function should suffice.

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi @AlphaPro06 

The PR looks great overall. Could you kindly make the requested changes? Also, the lint tests seem to be failing. Could you kindly fix that as well 😀

**Review by AlphaPro06 — COMMENTED** [review]
Made the requested changes, I don't understand the file in .idea/runConfiguration/_template__of_py_test.xml have the exact same file contente

**Review by AlphaPro06 — COMMENTED** [review]
I've made the requested changes, and it passed the lint test as well. Can you please approve the changes? 😃

**Comment by AlphaPro06** [comment]
Close #5493

**Review by AlphaPro06 — COMMENTED** [review]
Solved merge conflicts, hopefully now the code can merge. @YushaArif99 Please review and merge if my contribution looks good to you!

File: .idea/runConfigurations/_template__of_py_test.xml [hunk 5]
```diff
@@ -45,4 +45,4 @@
     <option name="_new_targetType" value="&quot;PATH&quot;" />
     <method v="2" />
   </configuration>
-</component>
```
Comment (YushaArif99) [inline .idea/runConfigurations/_template__of_py_test.xml:48]: There is an extra line at the end of this file. Might be easiest to rebase the fork and then push --force or something. Or you could simply copy and paste the contents of the file located in the main branch of ivy.

**Review by AlphaPro06 — COMMENTED** [review]
fixed the issues described

**Comment by AlphaPro06** [comment]
Close 5682

**Comment by AlphaPro06** [comment]
Close #5681

**Comment by AlphaPro06** [comment]
Close #5680

**Comment by AlphaPro06** [comment]
@YushaArif99 hi I made the changes you've requested, please let me know if there is anything else I would need to change. If there is nothing else for me to change, please approve and merge my pull request as soon as possible!

File: ivy/functional/frontends/tensorflow/raw_ops.py [hunk 6]
```diff
@@ -261,18 +256,10 @@ def Square(*, x, name="Square"):
 Sub = tf_frontend.subtract
 
 
-
 def Sum(*, input, axis, keep_dims=False, name="Sum"):
     return ivy.astype(ivy.sum(input, axis=axis, keepdims=keep_dims), input.dtype)
 
 
-def Tan(*, x, name="Tan"):
```
Comment (YushaArif99) [inline ivy/functional/frontends/tensorflow/raw_ops.py:269]: Why modify the contents of` Tf raw_ops.py file`? Could you kindly revert these changes if they were made by accident.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_namespace_functions.py [hunk 7]
```diff
@@ -938,3 +938,153 @@ def test_jax_numpy_fmax(
...
+        fn_tree="numpy.asarray",
+        x=x[0]
+    )
+
+
+# cumprod
+@handle_cmd_line_args
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("integer"), num_arrays=1
```
Comment (YushaArif99) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_namespace_functions.py:977]: Could you kindly explain as to why you're using "integers" and "floats" separately for `cumsum `and `cumprod `but using "valid" for `asarray`. Is valid not working with the other 2?

**Review by YushaArif99 — CHANGES_REQUESTED** [review]
Hi, @AlphaPro06. Could you kindly review the changes I have requested? Also, kindly resolve the merge conflicts so that the CI tests can run. I have tried running the tests on codespaces and they are failing for all 3 functions. Kindly look into it as well. Thanks!😊

**Comment by AlphaPro06** [comment]
@YushaArif99 Hi, I hope all is well. I tried to really debug the issues I have facing. So I tried using valid dtypes for both cumsum and cumprod. I really don't understand how else can I fix the Hypothesis errors I am recieving, I apologize but I really feel I don't know what else can I do to fix these errors. Do you have any suggestions that could help? Thank you!

Screen Shot 2022-10-16 at 11 59 52 PM
Screen Shot 2022-10-17 at 12 01 27 AM

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_namespace_functions.py (hunk 7 above)
Comment (AlphaPro06) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_namespace_functions.py]: Yeah "valid" wasn't really working on cumsum and cumprod, it's also why I tried using "integers" and "floats" differently but now I think "valid" is working now for cumsum and prod but I am getting the MultipleFailures exception where Hypothesis found 2 distinct failures I've mentioned in my new comment.
